### PR TITLE
Refactor Logs (and update LogViewer), new proposed format for Updates

### PR DESCRIPTION
**Pull request by kress95**
Yes, single commit 891 additions / deletions, and it will grow some more (tests aren't updated yet).

#### MAIN CHANGES

- Simplify `Game/Servers/Logs/Model.elm` **A LOT**
- Finally found an ideal update format for our structure. [1]
- Adapt `Game/Servers/Update.elm` to the new format.
- Rewrite LogViewer (and it's menu) to conform the new Game (still no format refactoring).
- Adds some new Dispatch code.

~~I'll try to update tests today. (maybe from home)~~ **DONE**

---

> #### [1]
>
> It boils down to **three simple rules**:
>
>   1. Never route and describe actions on a single function, split it over multiple ones instead.
>   2. Routing functions should be prefixed with `update`.
>   3. Action functions should be prefixed with `on`.
>
> Implicit rules:
>   - Try to keep your functions ordered, this makes it easier to navigate the file.
>   - Public functions should be kept on top, updates will usually contain just two public functions. (`update` and `bootstrap`)
>
> By doing that your update will reach the readeability level of `Servers` and `Logs` updates introduced here. Following the examples here will also simplify Models a lot.

---
This change is [Reviewable](https://reviewable.io/reviews/hackerexperience/heborn/200)

### CODE REVIEW

**Comment by kress95** [comment]
TODO: 

- [ ] Fix tests
- [ ] Add fixme from https://github.com/HackerExperience/HEBorn/pull/199
- [ ] Fix a problem with the dict reducer (and maybe try to make it generic)

**Comment by PedroHLC** [comment]
And that's it! PR #200 :awesome:
@renatomassaro I think its worth a pizza! :pizza:

_Sent from my Google Nexus 5 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by PedroHLC** [comment]
:lgtm:

---

Reviewed 25 of 25 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/heborn/200#-:-KrGyUsM9gcybwDmntLY:bnfp4nl)*

File: src/Apps/LogViewer/Messages.elm [hunk 1]
```diff
@@ -13,7 +13,7 @@ type Msg
     | ApplyEditing ID
     | LeaveEditing ID
     | StartCrypting ID
-    | StartUncrypting ID
```
Comment (PedroHLC) [inline src/Apps/LogViewer/Messages.elm]: You never saw this written here! Okay? 👀

**Pull request merged**